easybuild/easyblocks/x/xmipp.py

```diff
@@ -0,0 +1,116 @@
+##
+# Copyright 2009-2015 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://vscentrum.be/nl/en),
+# the Hercules foundation (http://www.herculesstichting.be/in_English)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# http://github.com/hpcugent/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing Xmipp, implemented as an easyblock
+
+@author: Jens Timmerman (Ghent University)
+"""
+import glob
+import os
+
+from easybuild.framework.easyblock import EasyBlock
+from easybuild.tools.filetools import mkdir, extract_file
+from easybuild.tools.modules import get_software_root, get_software_version
+from easybuild.tools.run import run_cmd
+
+
+class EB_Xmipp(EasyBlock):
+    """Support for building/installing Xmipp."""
+
+    def __init__(self, *args, **kwargs):
+        """Easyblock constructor, enable building in installation directory."""
+        super(self.__class__, self).__init__(*args, **kwargs)
+        self.build_in_installdir = True
+
+    def extract_step(self):
+        """Extract sources."""
+        # strip off 'xmipp' part to avoid having everything in a 'xmipp' subdirectory
+        self.cfg['unpack_options'] = "--strip-components=1"
+        super(self.__class__, self).extract_step()
+
+    def configure_step(self):
+        """Configure xmipp build via a provided wrapper around sconss."""
+        # install step expects these to exist
+        mkdir('lib')
+        mkdir('bin')
+        # run interactive 'config' script to generate install.info file
+        try:
+            # ln -s `which python` xmipp_python
```

Comment:
`# symlink xmipp_python to python binary`

the below is *not* equivalent to `which python`, rather you're relying on `Python` being specified as a dep

if it's not, this will go down hard btw, you want something like:

```python
python_root = get_software_root('Python')
if python_root:
    os.symlink(os.path.join(python_root, 'bin', 'python'), 'xmipp_python')
else:
    self.log.error("Python not loaded as a dependency, which is require for %s" % self.name)
```